lib/hex_web/throttle.ex

```diff
@@ -8,20 +8,23 @@ defmodule HexWeb.Throttle do
     GenServer.start_link(__MODULE__, new_state(opts), opts)
   end
 
-  def wait(pid, timeout \\ @timeout) do
+  def wait(pid, increment, timeout \\ @timeout) when increment >= 1 do
     try do
-      GenServer.call(pid, :run, timeout)
+      GenServer.call(pid, {:run, increment}, timeout)
     catch
       :exit, :timeout ->
         GenServer.cast(pid, {:cancel, self()})
         exit(:timeout)
     end
   end
 
-  def handle_call(:run, from, state) do
+  def handle_call({:run, increment}, _, %{rate: rate}) when increment > rate do
+    raise "increase must not exceed rate: #{inspect rate}, got: #{inspect increment}"
```

Comment:
otherwise the process "deadlocks". I tried unit test this, but I couldn't get it to work, I tried `catch_exit(Throttler.wait(...))` but no dice.